bench/wtperf/wtperf_misc.c

```diff
@@ -30,6 +30,51 @@
 
 #define WT_BACKUP_COPY_SIZE (128 * 1024)
 
+int
+delete_index_key(WTPERF *wtperf, WT_CURSOR *index_cursor, char *key_buf, uint64_t keyno)
+{
+    CONFIG_OPTS *opts;
+    uint64_t i, index_val;
+    size_t len;
+    int ret;
+
+    opts = wtperf->opts;
+    len = opts->key_sz + opts->value_sz_max;
+
+    /* Delete any other index entries. */
+    for (i = 1; i <= wtperf->index_max_multiplier; ++i) {
+        index_val = i * INDEX_BASE;
+        testutil_snprintf(key_buf, len, "%8" PRIu64 ":%11" PRIu64, index_val, keyno);
+        index_cursor->set_key(index_cursor, key_buf);
+        ret = index_cursor->remove(index_cursor);
+        if (ret == 0)
+            break;
+        if (ret == WT_NOTFOUND)
+            continue;
+        if (ret == WT_ROLLBACK)
+            return (ret);
+        lprintf(wtperf, ret, 1, "Delete earlier index key failed");
+    }
+    return (0);
+}
+
+/*
+ * Set up an index key based on global values.
+ */
+void
+generate_index_key(WTPERF *wtperf, u_int id_mult, char *key_buf, uint64_t keyno)
+{
+    CONFIG_OPTS *opts;
+    uint64_t index_val;
+    size_t len;
+
+    opts = wtperf->opts;
+    len = opts->key_sz + opts->value_sz_max;
+
+    index_val = id_mult * INDEX_BASE;
+    testutil_snprintf(key_buf, len, "%" PRIu64 ":%" PRIu64, index_val, keyno);
+}
+
```

Comment:
It looks to me like the index key for main table keyno is 20000:<keyno> for thread 0, 30000:<keyno> for thread 1, etc.  This does have the properties that there is some "spread" of index keys due to multiple threads, and index keys can be deleted by walking through the possible values.  But it does mean that index keys coming from a particular thread are clustered.  So this may have the properties needed for chasing the backup problem, but how is it for other benchmarking?  At minimum we need a comment explaining this characteristic.

There could be alternative ways to generate the index keys, we can discuss.

Reply:
I'll definitely add a comment. And yes, let's discuss your ideas. I made a simplistic index just to get the of deleting in one area and inserting in another happening. There is no delete operation/path in wtperf prior to this.